src/typeschema/types.ts

```diff
@@ -207,12 +207,39 @@ export interface ProfileConstraint {
     };
 }
 
+export interface FieldSlicing {
+    discriminator?: Array<{ type?: string; path: string }>;
+    rules?: string;
+    ordered?: boolean;
+    slices?: Record<string, FieldSlice>;
+}
+
+export interface FieldSlice {
+    min?: number;
+    max?: number;
+    match?: Record<string, unknown>;
+    required?: string[];
+    excluded?: string[];
+}
+
+export interface ExtensionSubField {
+    name: string;
+    url: string;
+    valueType?: Identifier;
+    min?: number;
+    max?: string;
+}
+
 export interface ProfileExtension {
```

Comment:
Looks like it is not a part of TypeSchema definition. Maybe define it internally in transformer or utils to avoid mistakes?